spec/requests/api/new_order_spec.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+RSpec.describe Api::OrdersController, type: :request do
+  let!(:product_1) { create(:product, name: 'Salad', price: 10) }
+  let!(:product_2) { create(:product, name: 'Ice Cream', price: 20) }
+  
+  before do
+    post '/api/orders', params: { product_id: product_1.id }
+    order_id = JSON.parse(response.body)['order']['id']
+    @order = Order.find(order_id)
+  end
+  
+  describe 'POST /api/orders' do
+    it 'responds with success message' do
+      expect(JSON.parse(response.body)['message']).to eq 'This item has been added to your order.'
+    end
+    it 'responds with order id' do
+      expect(JSON.parse(response.body)['order']['id']).to eq @order.id
```

Comment:
You should probably consider adding a helper to parse the response body to JSON and include it in your `rails_helper`

Reply:
Thank you for that tip, we have added that now and changed how the spec was written